### PR TITLE
server: Fix regression on /jobs/:id page.

### PR DESCRIPTION
**Pull request by benbardin**
Job details page is currently broken on master. Looks like the bug was introduced in 931957c171f1f3aa5b3d49462eb257aaf0c75f51. I added a test to protect against this going forward!

Release note: None

### CODE REVIEW

File: pkg/ccl/serverccl/admin_test.go [hunk 1]
```diff
@@ -93,3 +108,48 @@ func TestAdminAPIChartCatalog(t *testing.T) {
...
+	if len(backups) != 1 {
+		t.Errorf("Expected 1 Backup job, got %d", len(backups))
+	}
+
+	jobID := backups[0].ID
+
+	var jobRes serverpb.JobResponse
+	path := fmt.Sprintf("jobs/%v", jobID)
+	var err error
+	if err = getAdminJSONProto(s, path, &jobRes); err != nil {
```
Comment (stevendanna) [inline pkg/ccl/serverccl/admin_test.go:148]: No need to pre-declare err here:

```suggestion
	if err := getAdminJSONProto(s, path, &jobRes); err != nil {
```

**Review by stevendanna — COMMENTED** [review]
Thanks for jumping on this issue and adding a test. I left a few minor nitpicks on the test, but nothing major.

We often try to add a sentence or two about what the previous error was to the commit message.

**Comment by ajwerner** [comment]
Thanks!

**Comment by benbardin** [comment]
Updated the error handling, thank you for the suggestion!

**Review by stevendanna — APPROVED** [review]
Thanks!

**Pull request merged**